save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/ProjectStatisticMenu.kt

```diff
@@ -0,0 +1,113 @@
+@file:Suppress("FILE_NAME_MATCH_CLASS", "FILE_WILDCARD_IMPORTS", "LargeClass")
+
+package org.cqfn.save.frontend.components.basic
+
+import org.cqfn.save.agent.LatestExecutionStatisticDto
+import org.cqfn.save.domain.TestResultStatus
+import org.cqfn.save.frontend.components.tables.tableComponent
+import org.cqfn.save.frontend.externals.chart.DataPieChart
+import org.cqfn.save.frontend.externals.chart.pieChart
+import org.cqfn.save.frontend.externals.chart.randomColor
+import org.cqfn.save.frontend.utils.apiUrl
+import org.cqfn.save.frontend.utils.decodeFromJsonString
+import org.cqfn.save.frontend.utils.get
+import org.cqfn.save.frontend.utils.unsafeMap
+
+import org.w3c.fetch.Headers
+import react.*
+import react.dom.div
+import react.dom.td
+import react.table.columns
+
+/**
+ * ProjectStatisticMenu component props
+ */
+external interface ProjectStatisticMenuProps : Props {
+    /**
+     * Id of execution
+     */
+    var executionId: Long?
+
+    /**
+     * list of tests
+     */
+    var latestExecutionStatisticDtos: List<LatestExecutionStatisticDto>?
+}
+
+/**
+ * @return ReactElement
+ */
+@Suppress("TOO_LONG_FUNCTION", "LongMethod", "MAGIC_NUMBER")
+fun projectStatisticMenu() =
+        fc<ProjectStatisticMenuProps> { props ->
+
+            div("row justify-content-center") {
+                // ===================== LEFT COLUMN =======================================================================
+                div("col-2 mr-3") {
+                    div("text-xs text-center font-weight-bold text-primary text-uppercase mb-3") {
+                        +"Chart pie"
+                    }
+
+                    div("col-xl col-md-6 mb-4") {
+                        val data: MutableList<DataPieChart> = mutableListOf()
+                        props.latestExecutionStatisticDtos?.map {
+                            data.add(DataPieChart(it.testSuiteName, it.countTest, randomColor()))
+                        }
```

Comment:
Please either create an immutable list from `.map` or use `forEach` (I prefer the former)

Reply:
Fixed